src/include/connection.h

```diff
@@ -248,17 +248,18 @@ struct __wt_connection_impl {
 
     const char *cfg; /* Connection configuration */
 
-    WT_SPINLOCK api_lock;        /* Connection API spinlock */
-    WT_SPINLOCK checkpoint_lock; /* Checkpoint spinlock */
-    WT_SPINLOCK fh_lock;         /* File handle queue spinlock */
-    WT_SPINLOCK flush_tier_lock; /* Flush tier spinlock */
-    WT_SPINLOCK metadata_lock;   /* Metadata update spinlock */
-    WT_SPINLOCK reconfig_lock;   /* Single thread reconfigure */
-    WT_SPINLOCK schema_lock;     /* Schema operation spinlock */
-    WT_RWLOCK table_lock;        /* Table list lock */
-    WT_SPINLOCK tiered_lock;     /* Tiered work queue spinlock */
-    WT_SPINLOCK turtle_lock;     /* Turtle file spinlock */
-    WT_RWLOCK dhandle_lock;      /* Data handle list lock */
+    WT_SPINLOCK api_lock;               /* Connection API spinlock */
+    WT_SPINLOCK checkpoint_lock;        /* Checkpoint spinlock */
+    WT_RWLOCK debug_log_retention_lock; /* Log retention reconfiguration lock */
```

Comment:
As always naming suggestions are welcome.
The variables being protected are debug variables that control how much of the log we retain. We're specifically protecting them from access while they're in the process of being reconfigured.

Reply:
Perhaps `debug_ckpt_log_lock`? It isn't obvious to me that this lock given its current name would apply to debug_ckpt.

Reply:
In that case I'd prefer to change the naming of `debug_ckpt`. `debug_ckpt` just tracks the LSNs associated with the last N checkpoints, and we use this to determine the oldest log file we're not allowed to discard. Focusing on the checkpoint here feels like a distraction.

Reply:
We probably need another persons opinion here who is more familiar with these variables.